firestore-send-email/functions/src/helpers.ts

```diff
@@ -0,0 +1,24 @@
+import { createTransport, Transporter } from "nodemailer";
+import { URL } from "url";
+import { Config } from "./types";
+
+export const setSmtpCredentials = (config: Config) => {
+  const { smtpConnectionUri, smtpPassword } = config;
+  let url = new URL(smtpConnectionUri);
+  let smtpCredentials;
+  if (!url) {
+    return (smtpCredentials = null);
+  }
+  if (smtpConnectionUri) {
+    smtpCredentials = createTransport({
+      host: decodeURIComponent(url.hostname),
+      port: parseInt(url.port),
+      secure: url.protocol.includes("smtps") ? true : false, // true for 465, false for other ports
```

Comment:
I'm not too sure if this defines what secure connection is https://nodemailer.com/smtp/.

May be worth looking to see if this field is needed. Or could even be another config option (in the future?).